src/frontend/apps/drive/src/features/explorer/components/modals/share/WorkspaceShareModal.tsx

```diff
@@ -60,11 +62,27 @@ export const WorkspaceShareModal = ({
     [t]
   );
 
+  /**
+   * Keep previous search result when changing the query value to avoid flickering
+   * Because when it's the first time we have this value, the queryKey changes and by default react-query returns undefined
+   */
   const { data: users, isLoading: isLoadingUsers } = useUsers(
     { q: queryValue },
-    { enabled: !!queryValue && queryValue !== "" }
+    {
+      enabled: queryValue !== undefined && queryValue !== "",
+      initialData: previousSearchResult.current,
```

Comment:
It seems placeholderData ( previously keepPreviousData ) is made for this use case. WDYT about using this build-in mecanism, maybe it will reduce to complexity added ? https://tanstack.com/query/latest/docs/framework/react/guides/migrating-to-v5/#removed-keeppreviousdata-in-favor-of-placeholderdata-identity-function